modules/qtwidgets/rawdatareaderdialogqt.cpp

```diff
@@ -43,27 +44,103 @@ namespace inviwo {
+                valueRangeMax_->setValue(255);
+                break;
+            case 1:  // CHAR
+                dataRangeMin_->setValue(-128);
+                dataRangeMax_->setValue(127);
+                valueRangeMin_->setValue(-128);
+                valueRangeMax_->setValue(127);
+                break;
+            case 2:  // USHORT
+                dataRangeMin_->setValue(0);
+                dataRangeMax_->setValue(std::numeric_limits<glm::u16>::max());
+                valueRangeMin_->setValue(-1000);
+                valueRangeMax_->setValue(3000);
+                break;
+            case 3:  // USHORT (12-bits)
+                dataRangeMin_->setValue(0);
+                dataRangeMax_->setValue(4095);
+                valueRangeMin_->setValue(-1000);
+                valueRangeMax_->setValue(3000);
+                break;
+            case 4:  // UINT
+            case 5:  // INT
+            case 6:  // FLOAT
+            case 7:  // DOUBLE
+				dataRangeMin_->setValue(0);
+				dataRangeMax_->setValue(1);
+				valueRangeMin_->setValue(0);
+				valueRangeMax_->setValue(1);
+                break;
+		};
+    });
     QLabel* channelLabel = new QLabel("Data channels");
     channels_ = new QSpinBox();
-    channels_->setRange(0, 16);
+    channels_->setRange(1, 4);
     channels_->setValue(1);
-    dataTypeLayout->addWidget(bitDepthLabel, 0, 0);
-    dataTypeLayout->addWidget(bitDepth_,     0, 1);
-    dataTypeLayout->addWidget(channelLabel,  1, 0);
-    dataTypeLayout->addWidget(channels_,     1, 1);
+
+    QLabel* dataRangeLabel = new QLabel("Data format range");
+	dataRangeLabel->setToolTip(
+		"Data range refer to the range of the data type, i.e. [0 4095] for 12-bit unsigned integer "
+		"data.");
+	dataRangeMin_ = new QDoubleSpinBox();
+    dataRangeMin_->setRange(-std::numeric_limits<double>::max(),
+                            std::numeric_limits<double>::max());
+    dataRangeMax_ = new QDoubleSpinBox();
+    dataRangeMax_->setRange(-std::numeric_limits<double>::max(),
+                            std::numeric_limits<double>::max());
+    dataRangeMax_->setValue(255.0);
+    QLabel* valueRangeLabel = new QLabel("Value range");
+    valueRangeLabel->setToolTip("Value range refer to the physical meaning of the value, i.e. Hounsfield value range for human tissue [-1000 3000]");
+	valueRangeMin_ = new QDoubleSpinBox();
+	valueRangeMin_->setRange(-std::numeric_limits<double>::max(), std::numeric_limits<double>::max());
+	valueRangeMax_ = new QDoubleSpinBox();
+	valueRangeMax_->setRange(-std::numeric_limits<double>::max(), std::numeric_limits<double>::max());
+	QLabel* unit = new QLabel("Unit (m/s, HU, W)");
+	valueUnit_ = new  QLineEdit();				///< Unit, i.e. Hounsfield/absorption/W.
+
+	auto rowCount = 0;
+    dataTypeLayout->addWidget(bitDepthLabel, rowCount, 0);
+    dataTypeLayout->addWidget(bitDepth_, rowCount++, 2);
+
+	dataTypeLayout->addWidget(new QLabel("Min"), rowCount, 1);
```

Comment:
are you using tabs instead of spaces here?

Reply:
There are many places att github reports tabs rather than spaces.

Reply:
Wonder if VS has changed my setting. Will fix